### PR TITLE
[Perf][Multimodal] Avoid duplicate text embedding in Qwen2.5-Omni

### PR DESCRIPTION
**Pull request by waizuichougou**
# [Perf][Multimodal] Avoid duplicate text embedding in Qwen2.5-Omni

## Purpose

`Qwen2_5OmniThinkerForConditionalGeneration.embed_input_ids` embeds the text
input before it knows which multimodal merge path is required. For a regular
multimodal request, the method then calls the parent implementation, which
embeds the same text input again. For an audio-in-video request, the custom
interleaved path also embeds the text input again. The first embedding result
is discarded in both cases. This is a performance-only change and does not
alter multimodal embedding placement behavior.

This change determines whether interleaved audio/video merging is required
before embedding text. The regular path delegates directly to the parent
implementation, while the interleaved path performs its custom merge with
one text embedding. The empty-multimodal path also delegates to the parent
implementation without changing the resulting tensor.

### Duplicate-work check

The open pull-request list and recent multimodal performance changes were
checked for work covering duplicate text embedding in
`Qwen2_5OmniThinkerForConditionalGeneration.embed_input_ids`. The nearby
logits-state caching optimization addresses a different sampling-stage
hotspot, so this change does not duplicate it.

### Validation

```bash
python -m pytest -q tests/models/multimodal/processing/test_qwen2_5_omni_embed.py
ruff check vllm/model_executor/models/qwen2_5_omni_thinker.py \
    tests/models/multimodal/processing/test_qwen2_5_omni_embed.py
ruff format --check vllm/model_executor/models/qwen2_5_omni_thinker.py \
    tests/models/multimodal/processing/test_qwen2_5_omni_embed.py
git diff --check
```

The existing multimodal regression test is extended with call-count assertions
for the standard and audio-in-video paths. No new test file is added.

### Qwen2.5-Omni embedding-stage benchmark

The actual vLLM Qwen2.5-Omni model class was benchmarked on an NVIDIA A100.
The language embedding used the Qwen2.5-Omni-3B vocabulary size (151,936),
hidden size (2,048), and `bfloat16` dtype. This isolates the changed
`embed_input_ids` stage while keeping the model's real token dimensions and
multimodal merge logic. The benchmark does not include the vision/audio
encoders or language-model layers, which are unchanged by this PR.

For each input shape, 30 warm-up calls were discarded. Nine timed samples were
collected, each averaging 30 calls, and the table reports the median time per
call. The unmodified and patched versions used the same inputs and settings.

### Existing tests and checks

- `tests/models/multimodal/processing/test_qwen2_5_omni_embed.py`: **15 passed**
- `ruff check`: passed
- `ruff format --check`: passed
- `git diff --check`: passed

### Qwen2.5-Omni embedding-stage benchmark

`Change` is `(patched / unmodified - 1)`; negative values are improvements.
The embedding-call column is measured per `embed_input_ids` invocation.

| Input tokens | Multimodal tokens | Unmodified (µs) | Patched (µs) | Change | Text embedding calls |
| ---: | ---: | ---: | ---: | ---: | ---: |
| 128 | 32 | 189.542 | 153.088 | **-19.2%** | 2 → 1 |
| 512 | 64 | 190.635 | 155.307 | **-18.5%** | 2 → 1 |
| 1,024 | 256 | 199.612 | 161.519 | **-19.1%** | 2 → 1 |
| 2,048 | 512 | 206.063 | 170.018 | **-17.5%** | 2 → 1 |
| 4,096 | 1,024 | 238.967 | 198.417 | **-17.0%** | 2 → 1 |
| 8,192 | 2,048 | 315.972 | 254.839 | **-19.3%** | 2 → 1 |

The result is consistent across short and long multimodal sequences. The
existing output-placement assertions remain unchanged, while the discarded
duplicate embedding is removed from every non-interleaved and interleaved
multimodal call.

### AI-assisted contribution

This contribution was developed with AI assistance.

### CODE REVIEW

**Comment by DarkLight1337** [comment]
@claude review

File: tests/models/multimodal/processing/test_qwen2_5_omni_embed.py [hunk 1]
```diff
@@ -196,14 +196,14 @@ def fake_lm_embed(ids: torch.Tensor) -> torch.Tensor:
         return ids.float().unsqueeze(-1).expand(-1, hidden).clone()
 
     lang_model = Mock()
-    lang_model.embed_input_ids = fake_lm_embed
+    lang_model.embed_input_ids = Mock(side_effect=fake_lm_embed)
```
Comment (DarkLight1337) [inline tests/models/multimodal/processing/test_qwen2_5_omni_embed.py]: The test additions are honestly unnecessary. Let's just make the model change

**Review by DarkLight1337 — APPROVED** [review]
Thanks
